### PR TITLE
Closes #2028 removed erroneous section. updating rules section

### LINKED ISSUE #2028: Documentation:  Imputation vignette needs a "rephrase" around preserving partial dates and highest imputation
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

The imputation vignette says that we do not allow users to pick any parts of the partial date.  However, we have a `preserve` argument that does that!!  We should remove this from the vignette.

image

We should open up the vignette with discussion on the `highest_imputation` this could just be a direct copy of the roxygen documents.  This should be discussed in the first example.

### PR DESCRIPTION
**Pull request by bms63**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/imputation.Rmd [hunk 1]
```diff
@@ -28,31 +45,19 @@ Relative Day) are numeric variables. They can be derived only if the date or
...
 
-In {admiral} we use only two functions `derive_vars_dt()` and
-`derive_vars_dtm()` for date and datetime imputations respectively. In all other
-functions where dates can be passed as an argument, we expect full dates or
-datetimes (unless otherwise specified), so if any possibility of partials then
-these functions should be used as a first step to make the required imputation.
+In `{admiral}` users will primarily use two functions `derive_vars_dt()` and `derive_vars_dtm()` for date and datetime imputations respectively. In all other functions where dates can be passed as an argument, we expect full dates or datetimes (unless otherwise specified), so if any possibility of partials then these functions should be used as a first step to make the required imputation.
 
-## Required Packages
+The functions that need to do date/time imputation follow a rule that we have called **Highest Imputation**, which has a corresponding argument in all our functions called `highest_imputation`. This rule allows a user to control which components of the DTC value are imputed if they are missing.
```
Comment (bundfussr) [inline vignettes/imputation.Rmd:50]: I think we should not only mention that this rule exists but also explain what it means.

File: vignettes/imputation.Rmd [hunk 2]
```diff
@@ -18,6 +18,23 @@ library(admiraldev)
 
 # Introduction
 
+This vignette is broken into three major sections.  The first section briefly explores 
+the imputation rules used in `{admiral}`.  The second section focuses on imputation functions
+that work on vectors with lots of small examples to explore the imputation rules. These **vector-based** functions form the backbone of `{admiral}`'s more powerful functions `derive_vars_dt()` and `derive_vars_dtm()` for building ADaM dataset. The final section moves into more detailed examples that a user might  face while working on ADaMs in need of `---DT` and `---DTM` variables.
```
Comment (zdz2101) [inline vignettes/imputation.Rmd:23]:
```suggestion
that work on vectors with lots of small examples to explore the imputation rules. These **vector-based** functions form the backbone of `{admiral}`'s more powerful functions `derive_vars_dt()` and `derive_vars_dtm()` for building ADaM dataset. The final section moves into more detailed examples that a user might face while working on ADaMs in need of `---DT` and `---DTM` variables. 
```

File: vignettes/imputation.Rmd [hunk 3]
```diff
@@ -28,31 +45,19 @@ Relative Day) are numeric variables. They can be derived only if the date or
...
+In `{admiral}` users will primarily use two functions `derive_vars_dt()` and `derive_vars_dtm()` for date and datetime imputations respectively. In all other functions where dates can be passed as an argument, we expect full dates or datetimes (unless otherwise specified), so if any possibility of partials then these functions should be used as a first step to make the required imputation.
 
-## Required Packages
+The functions that need to do date/time imputation follow a rule that we have called **Highest Imputation**, which has a corresponding argument in all our functions called `highest_imputation`. This rule allows a user to control which components of the DTC value are imputed if they are missing. 
 
-The examples of this vignette require the following packages.
+The default imputation for `_dtm()` functions, e.g. `impute_dtc_dtm()`, `derive_vars_dtm()`, is "h" (hours). A user can specify that that no imputation is to be done by setting `highest_imputation = n`. However, for for `_dt()` functions, e.g. `impute_dtc_dt()`, `derive_vars_dt()` the default imputation is already set as `highest_imputation = n`.
```
Comment (zdz2101) [inline vignettes/imputation.Rmd:52]:
```suggestion
The default imputation for `_dtm()` functions, e.g. `impute_dtc_dtm()`, `derive_vars_dtm()`, is "h" (hours). A user can specify that that no imputation is to be done by setting `highest_imputation = n`. However, for for `_dt()` functions, e.g. `impute_dtc_dt()`, `derive_vars_dt()` the default imputation is already set as `highest_imputation = "n"`.
```

**Review by zdz2101 — COMMENTED** [review]
Super small stuff, primarily see stefan's comment

**Review by zdz2101 — APPROVED** [review]

File: vignettes/imputation.Rmd (hunk 1 above)
Comment (bms63) [reply vignettes/imputation.Rmd]: I put in a pretty weak argument.  

I started writing and then making a table and was left with more questions then answers.  I think whatever we make in this vignette to help explain the rule we should then populate into the function documentation as I was getting so confused on how to explain it.  Unless you have a pithy phrase?

Going to make a new issue for this update

Comment (bms63) [reply vignettes/imputation.Rmd]: https://github.com/pharmaverse/admiral/discussions/2090

**Pull request merged**